### DISCUSSION TITLE
Fifth Community Meeting 25/8/2022

### DISCUSSION THREAD

**Discussion by VojtechBartoska** [espressif/arduino-esp32#7167] [General]
Hello to all Arduino ESP32 Core Users 🚀 

**Meeting purpose,**
we appreciate all of your contribution and help and we would like to improve your User experience on this Project. 🤩

As a part of this improvement we would like to uncover our internal Development to you, have clear and transparent planning and also listen to your requests, ideas and valuable Feedback.

To fulfill this, we would like to held every Month Community Meetings where we can share main updates and also have an open discussion with you.

---

**💥Fifth Community Meeting will take place on Thursday 25/08/2022 at 4 pm GMT+2 time zone. (1 hour length) We will use Google Meet.**

---

We want to have some idea about how many of you would be interested in joining these Meetings so instead of sharing directly link to Online Meeting itself, it will work by invitation.

**If you want to be added to upcoming Meeting and receive invitation please send an email to [vojtech.bartoska@espressif.com](mailto:vojtech.bartoska@espressif.com).** 

If you participated on any Community Meeting, you are automatically invited to upcoming Meeting and it's up to you if you joins us again or not.

---

### 1) Espressif Introduction (2 minutes) 
Meeting purpose & rules
### 2) Development updates (5 minutes)
Summary of our current development and news
#### Reorganizing Boards in Arduino IDE
- more context in the comments below
### 3) Open Discussions
Place for any topic which is not in Agenda.

See you on the Meeting!

**Comment by VojtechBartoska** [comment]
## Reorganizing board menu

**Motivation**
There are 160 ESP boards (117 ESP32) - that's a very long list to scroll through - let's reorganize it to make it easier to find your board and save you from endless scrolling.

---
According to [response on Arduino forum](https://forum.arduino.cc/t/tools-board-esp32-new-submenu/1022376) it is not possible to create a submenu directly under the option `ESP32 Arduino` such as `Tools > Board > ESP32 Arduino > submenu > boards`.

---
**Options on how to reorganize the Board menu:**
1. It is possible to create a new item next to the Flashing options such as Upload speed, CPU frequency, debug level, etc.

2. Another idea is to generate multiple release packages. We could have packages grouped by SoCs as ESP32, ESP32-S3...

---

Please share your ideas here and let's figure out how to make everyone's life easier.

**Reply by PilnyTomas** [reply]
If we split the top level-folders the menu could look something like this:
![menu](https://user-images.githubusercontent.com/34927466/186133495-dceea976-20f5-491d-b36c-3feaee25983d.png)
There are still 117 ESP32 boards :(
This split would require to rearrange some files and create links.
From original folder structure `~/Arduino/hardware/esp32` where the `esp32` is the project top folder we would need to change the top folder to `espressif` and split the rest to:
```
espressif/
├── esp32
├── esp32-c3
├── esp32-s2
└── esp32-s3
```

**Reply by PilnyTomas** [reply]
Checkout the proposition in PR #7172

**Reply by per1234** [reply]
>It is possible to create a new item next to the Flashing options such as Upload speed, CPU frequency, debug level, etc.

Why wasn't this option included in the meeting slides @VojtechBartoska?

**Reply by per1234** [reply]
>split the top level-folders

What you are doing here is creating four separate boards platforms. If you are only installing them manually by cloning to the `hardware` subfolder of your sketchbook, you might miss that each of these platforms must be installed independently via the Arduino Boards Manager.

So it is option two ("generate multiple release packages").

**Comment by VojtechBartoska** [comment]
Slides: 
[Fifth_Community_meeting-Arduino-August_2022 (1).pdf](https://github.com/espressif/arduino-esp32/files/9679126/Fifth_Community_meeting-Arduino-August_2022.1.pdf)
